libraries/SITL/SIM_JSON.cpp

```diff
@@ -318,6 +345,8 @@ void JSON::recv_fdm(const struct sitl_input &input)
         airspeed_pitot = constrain_float(velocity_air_bf * Vector3f(1.0f, 0.0f, 0.0f), 0.0f, 120.0f);
     }
 
+    memcpy(rpm, state.motor.rpm.data, state.motor.rpm.length*sizeof(float));
```

Comment:
this is unsafe, it allows buffer-overflows. Needs a bounds check of the dest buffer.
```suggestion
    memcpy(rpm, state.motor.rpm.data, MIN(state.motor.rpm.length*sizeof(float), sizeof(rpm)));
```

Reply:
Fixed it.